velox/common/caching/tests/SsdFileTest.cpp

```diff
@@ -234,6 +234,30 @@ class SsdFileTest : public testing::Test {
     readAndCheckPins(pins);
   }
 
+  // Reads back the found entries and check their contents, return the number of
+  // entries found.
+  int32_t checkEntries(const std::vector<TestEntry>& entries) {
+    int32_t numFound = 0;
+    for (auto& entry : entries) {
+      std::vector<CachePin> pins;
+      pins.push_back(cache_->findOrCreate(
+          RawFileCacheKey{fileName_.id(), entry.key.offset},
+          entry.size,
+          nullptr));
+      if (pins.back().entry()->isExclusive()) {
```

Comment:
What if the pinned entry is not exclusive? Thanks!

Reply:
I asked the same question while reading the code `TEST_F(SsdFileTest, writeAndRead)` :)

My guess is it serves as a write lock. While it loads content from SsdPin to CachePin, it doesn't want someone else is also CachePin.